### DISCUSSION TITLE
How to prevent outdated builds in development

### DISCUSSION THREAD

**Discussion by steveswork** [vercel/next.js#16490] [Help]
When I make code changes in the server (e.g. a console.log statement) and ran `npm run dev`, the console.log would not display on the console.

But when I remove the console.log and run `npm run dev`, the console.log is displayed. I’ve also attempted to use `rimraf` to remove the entire .next folder before running the `npm run dev`. 

Though the folder was removed and then re-created after the build, the supposedly removed console.log still appeared on the console.

How can I prevent this behavior? It Is currently confounding the development experience.

Fwiw, I’m also using:
next connected custom express server
the getInitialProps static method
Next.js version: 9.3.0
OS: Windows 10

**Comment by timneutkens** [comment]
Are you on the latest version of Next.js? Also what OS?

Also getInitialProps runs in the browser when navigating.

**Comment by steveswork** [comment]
@timneutkens thank you for asking about the OS and the version info. I am currently using next 9.3.0 on Windows 10. I am also using the next backed custom express server. My original post has now been updated with those info as well.

I am using the getInitialProps really because it runs in the server and in the browser when navigating.
